### DISCUSSION TITLE
Using Embed instead of Packr

### DISCUSSION THREAD

**Discussion by paganotoni** [gobuffalo/buffalo#2129] [Ideas]
One of the foundational libraries in Buffalo is Packr. Packr allows buffalo applications to embed assets and other runtime dependencies into the compiled binary and has become an important tool not only for apps powered by Buffalo but also by the libraries and tools that Buffalo depends on.

While Packr has provided the foundation for a lot of the things that Buffalo is capable of In Go 1.16 the Go team provided the Embed standard library, which is integrated to the Go tooling in a way that we just have to use some build tags and the rest is handled by the standard go tooling. 

It seems like using the `embed` package is the right movement but wanted to put this out because the amount of work to solve this is not trivial and is worth reading comments and concerns of it. The areas I've thought this movement would imply are:

- CLI internals
- CLI generated code (breaking)

Thoughts on this?

cc @gobuffalo/core-managers @markbates

**Comment by fasmat** [comment]
I would welcome the migration from packr to embed. Regarding the efforts I cannot yet estimate the effort for it.
Some thoughts I had regarding the topic:

- ``buffalo dev`` watches *.go files for changes and rebuilds the app binary if necessary afaik. If any other files (that would be handled by packr/embed) change this might not trigger a rebuild that afaik would be necessary to work with embed.
- Interoperability with webpack / yarn / npm might need rework for both ``buffalo dev`` and a built binary.
- ``github.com/gobuffalo/suite`` relies heavily on packr
- ``github.com/gobuffalo/packd`` is already a transitional package between packr v1 and v2
- There is also ``github.com/markbates/pkger`` which I think was at some point meant to supersede packr, did that ever happen?
  - On that note: will grift also at some point be added to gobuffalo?
